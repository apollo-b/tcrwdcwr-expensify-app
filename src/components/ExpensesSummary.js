import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        {
            `Viewing
            ${props.expenses.length}
            ${props.expenses.length === 1 ? 'expense' : 'expenses'}
            totaling 
            ${numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}`
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);

// 2. Commit and Deploy
// -----

// Get the feature live!