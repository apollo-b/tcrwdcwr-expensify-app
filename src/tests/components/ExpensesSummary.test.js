import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should display correctly for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should display correctly for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});