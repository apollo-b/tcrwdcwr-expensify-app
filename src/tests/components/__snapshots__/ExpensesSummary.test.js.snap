// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correctly for 1 expense 1`] = `
<div>
  <h1>
    Viewing
            1
            expense
            totaling 
            $1.95
  </h1>
</div>
`;

exports[`should display correctly for multiple expenses 1`] = `
<div>
  <h1>
    Viewing
            3
            expenses
            totaling 
            $1,141.95
  </h1>
</div>
`;
